### PR TITLE
add option for deserializing binary into Buffer object

### PR DESCRIPTION
**Pull request by msmiley**
**promoteBuffers** {Boolean, default:false}

### CODE REVIEW

**Comment by christkv** [comment]
would need to be also applied to bson-ext module in c++ to be pulled in and include tests.

File: lib/bson/bson.js [hunk 1]
```diff
@@ -1282,8 +1284,13 @@ BSON.deserialize = function(buffer, options, isArray) {
           if(subType == Binary.SUBTYPE_BYTE_ARRAY) {
             binarySize = buffer[index++] | buffer[index++] << 8 | buffer[index++] << 16 | buffer[index++] << 24;
           }
-          // Slice the data
-          object[name] = new Binary(buffer.slice(index, index + binarySize), subType);
+          if(nativeBuffers) {
+            // Slice the data and create a native Buffer object
+            object[name] = new Buffer(buffer.slice(index, index + binarySize));
```
Comment (tellnes) [inline lib/bson/bson.js]: `buffer.slice()` returns a buffer, why not just use that? Now you are doing a memory copy. https://github.com/nodejs/io.js/blob/master/lib/buffer.js#L96-L100

Comment (msmiley) [inline lib/bson/bson.js]: you are absolutely right, thanks for catching that!

**Comment by msmiley** [comment]
sorry for taking so long on this, rebased and changed the name of the variable to promoteBuffers to be more in line with the existing promoteLongs option. Also, nativeBuffers may make some users think C++.

Added a couple tests, let me know if I need to add a test somewhere else.

Will make a PR on bson-ext.

**Pull request merged**